distribution/resources/bin/gateway.bat

```diff
@@ -17,151 +17,235 @@ REM  limitations under the License.
-		powershell -Command "Rename-Item -path \"%GWHOME%\runtime\bin\ballerina_1.bat\" -newName ballerina.bat"
-		CD "%GWHOME%"
-		for /f "skip=3 tokens=2 delims=:" %%A in ('powershell -command "get-host"') do (
-			set /a n=!n!+1
-			set c=%%A
-			if !n!==1 set PSVersion=!c!
-		)
-		set PSVersion=!PSVersion: =!
-		if %verbose%==T echo [%date% %time%] DEBUG: PowerShell version !PSVersion! detected!
-		set PSVersion=!PSVersion:~0,1!
-		echo [%date% %time%] Starting Micro-Gateway
-		IF !PSVersion! LEQ 3 (
-			echo [%date% %time%] Starting Micro-Gateway >>  .\logs\microgateway.log
-			call powershell ".\runtime\bin\ballerina run %BAL_ARGS% --api.usage.data.path=\"%usage_data_path%\" --b7a.http.accesslog.path=\"%unix_style_path%\" --b7a.config.file=".\conf\micro-gw.conf" | out-file -encoding ASCII -filepath .\logs\microgateway.log -Append"
-		 ) else (
-			REM For powershell version 4 or above , We can use `tee` command for output to both file stream and stdout (Ref: https://en.wikipedia.org/wiki/PowerShell#PowerShell_4.0)
-			call powershell ".\runtime\bin\ballerina run %BAL_ARGS% --api.usage.data.path=\"%usage_data_path%\" --b7a.http.accesslog.path=\"%unix_style_path%\" --b7a.config.file=".\conf\micro-gw.conf" | tee -Append .\logs\microgateway.log"
-		)
-	)
-:end
-goto endlocal
-
-:endlocal
+    DEL /Q /F %CONF_OUT_FILE%
+
+:continueInit
+    REM Change the windows style `\` path separator to unix style `/path/to/file` for log file path
+    SET "separator=/"
+    SET log_path="%GW_HOME%\logs\access_logs"
+    CALL SET ACCESS_LOG_PATH=%%log_path:\=%separator%%%
+
+    REM Do the same for analytics data file path
+    SET usage_path=%GW_HOME%\api-usage-data
+    CALL SET USAGE_DATA_PATH=%%usage_path:\=%separator%%%
+
+    CALL :startGateway %*
+
+    GOTO END
+
+REM -----------------------------------------------------------------------------
+REM --- END OF MAIN PROGRAM LOGIC ---
+REM -----------------------------------------------------------------------------
+
+REM -----------------------------------------------------------------------------
+REM --- START OF FUNCTION DEFINITION ---
+REM -----------------------------------------------------------------------------
+
+REM Start the gateway using internal ballerina distribution as the runtime
+:startGateway
+    REM Check if powershell is available
+    SET JAVA_ARGS=-Xms%JAVA_XMS_VALUE% -Xmx%JAVA_XMX_VALUE% -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=%GW_HOME%\heap-dump.hprof
+    WHERE POWERSHELL >NUL 2>NUl
+
+    IF %ERRORLEVEL% NEQ 0 (
+        ECHO WARN: Can't find powershell in the system!
+        ECHO WARN: STDERR and STDOUT will be piped to %GW_HOME%\logs\microgateway.log
+        "%JAVA_HOME%\bin\java.exe" %JAVA_ARGS% -Dmgw-runtime.home=%GW_HOME% -Dballerina.home=%GW_HOME%/runtime -jar "%EXEC_FILE%" %BAL_ARGS% --api.usage.data.path=%USAGE_DATA_PATH%  --b7a.http.accesslog.path=%ACCESS_LOG_PATH% --b7a.config.file="%GW_HOME%\conf\micro-gw.conf" >> "%GW_HOME%\logs\microgateway.log" 2>&1
```

Comment:
Needs to have first element of the Args to be in here and the rest of the args to be in the end.
Needs to change the shell script as well.
Like this.
`$JAVACMD ...
    -jar **${args[0]}** --api.usage.data.path=$GW_HOME/api-usage-data --b7a.http.accesslog.path=$GW_HOME/logs/access_logs --b7a.config.file=$GW_HOME/conf/micro-gw.conf **${args[@]:1}**  2>&1 | tee -a $GW_HOME/logs/microgateway.log
`